### PR TITLE
[v6] Backport: dronegen: drone config generator

### PR DESCRIPTION
**Pull request by webvictim**
Backports these PRs to `branch/v6`:

- #6071
- #6106 

Also updates e ref to account for https://github.com/gravitational/teleport.e/pull/242

This is needed to be able to release ARM/ARM64 RPM/DEB packages as part of 6.1.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -641,3 +658,16 @@ update-vendor:
...
 	rm -r vendor/github.com/gravitational/teleport/api
 	ln -s -r $(shell readlink -f api) vendor/github.com/gravitational/teleport
+
+# update-webassets updates the minified code in the webassets repo using the latest webapps
+# repo and creates a PR in the teleport repo to update webassets submodule.
+.PHONY: update-webassets
+update-webassets: WEBAPPS_BRANCH ?= 'master'
+update-webassets: TELEPORT_BRANCH ?= 'master'
+update-webassets:
+	build.assets/webapps/update-teleport-webassets.sh -w $(WEBAPPS_BRANCH) -t $(TELEPORT_BRANCH)
```
Comment (awly) [inline Makefile:668]: Missing this actual script from https://github.com/gravitational/teleport/pull/5853/files

Comment (webvictim) [reply Makefile:668]: Good catch, cherry-picked it

**Review by awly — APPROVED** [review]

File: dronegen/main.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		os.Exit(1)
+	}
+
+	if err := signDroneConfig(); err != nil {
+		fmt.Println("failed signing .drone.yml:", err)
+		os.Exit(1)
+	}
+}
+
+func writePipelines(path string, newPipelines []pipeline) error {
```
Comment (a-palchikov) [inline dronegen/main.go:37]: I've also run into an issue where I accidentally duplicated a step in a pipeline and the drone's error message is not super helpful:
```
Linter: duplicate step name
```
it even had the wrong pipeline name in the front. I was wondering whether you guys considered adding basic linting functionality prior to writing pipelines to disk? `drone lint` alone is not helpful - it fails locally for multiple reasons but I've seen linting implemented as part of [boilr-runner](https://github.com/drone/boilr-runner) (which we're using as well somewhere, probably indirectly).

**Review by a-palchikov — APPROVED** [review]

File: build.assets/Makefile [hunk 3]
```diff
@@ -82,30 +90,40 @@ build-binaries-fips: buildbox-fips
 #
 .PHONY:buildbox
 buildbox:
-	docker build \
-		--build-arg UID=$(UID) \
-		--build-arg GID=$(GID) \
-		--build-arg RUNTIME=$(RUNTIME) \
-		--cache-from $(BUILDBOX) \
-		--tag $(BUILDBOX) .
+	if [[ "$(BUILDBOX_NAME)" == "$(BUILDBOX)" ]]; then \
```
[the remark this answers is not in the document]
Comment (webvictim) [reply build.assets/Makefile:93]: That's odd. It seems that busybox `/bin/sh` implements `[[` which is why we haven't run into this before.

File: dronegen/main.go (hunk 2 above)
Comment (webvictim) [reply dronegen/main.go:37]: It sounds like a good idea to check that the list of output pipeline names doesn't have any duplicates.

**Comment by webvictim** [comment]
@russjones Please bot approve.

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**